### DISCUSSION TITLE
SOCKS proxy

### DISCUSSION THREAD

**Discussion by epoberezkin** [markqvist/Reticulum#507] [Ideas]
Is there a plan to implement SOCKS proxy so other apps could connect to Reticulum addresses, in the same way they connect to .onion addresses via SOCKS?

To make it useful I'd love to see it:
- supporting isolate by auth feature (so each set of SOCKS credentials would make client use a different Reticulum address).
- supporting Tor as well, so the same SOCKS proxy can be used for both Reticulum and Tor addresses.

What will it take (in terms of time/budget) for it to happen / to be prioritised?

**Comment by markqvist** [comment]
I think something like this should be created as an standalone program / utility *using* Reticulum, and not being part of the core protocol or `rns` package itself.

It would be a pretty simple and easy thing to code up.
